source/dofs/dof_tools.cc

```diff
@@ -1733,19 +1733,13 @@ namespace DoFTools
     // this subdomain id
 #ifdef DEBUG
     {
-      bool found = false;
-      for (typename Triangulation<
-             DoFHandlerType::dimension,
-             DoFHandlerType::space_dimension>::active_cell_iterator cell =
-             dof_handler.get_triangulation().begin_active();
-           cell != dof_handler.get_triangulation().end();
-           ++cell)
-        if (cell->subdomain_id() == subdomain)
-          {
-            found = true;
-            break;
-          }
-      Assert(found == true,
+      const bool cell_found = std::any_of(
+        dof_handler.begin_active(),
+        typename DoFHandlerType::active_cell_iterator{dof_handler.end()},
+        [subdomain](const typename DoFHandlerType::cell_accessor &cell) {
+          return cell.subdomain_id() == subdomain;
+        });
+      Assert(cell_found == true,
```

Comment:
Just inline the `std::any_of` into the assertion here, so that it doesn't have to run in release mode.

Reply:
Good idea! then we can remove the `#ifdef DEBUG` guard here.

Reply:
Nice -- I missed that!